### PR TITLE
Multiple formatter

### PR DESCRIPTION
**Pull request by kokspflanze**
Hi,

i changed the formatter system, so it is now possible to add more than 1 formatter, also changed the setter methods in `AbstractColumn` they return now `$this`, so it is now possible todo smth like 

```php
$foo->setMethod('bar')->setSecoundMethod('baz');
```

### CODE REVIEW

File: src/ZfcDatagrid/PrepareData.php [hunk 1]
```diff
@@ -222,18 +222,19 @@ public function prepare()
                 /*
                  * Custom formatter
                  */
-                if ($col->hasFormatter($this->getRendererName()) === true) {
-                    /* @var $formatter \ZfcDatagrid\Column\Formatter\AbstractFormatter */
-                    $formatter = $col->getFormatter($this->getRendererName());
-                    $formatter->setRowData((array) $row);
-                    $formatter->setRendererName($this->getRendererName());
+                if ($col->hasFormatter()) {
+                    $formatterList = $col->getFormatter();
```
Comment (ThaDafinser) [inline src/ZfcDatagrid/PrepareData.php]: If you use it only once, do the call directly in `foreach()` so you save a $var

File: src/ZfcDatagrid/Column/AbstractColumn.php [hunk 2]
```diff
@@ -60,11 +60,21 @@
 
     protected $rendererParameter = [];
 
+    /**
+     * @var AbstractFormatter[]
+     */
     protected $formatter;
```
Comment (ThaDafinser) [inline src/ZfcDatagrid/Column/AbstractColumn.php]: $formatters

File: src/ZfcDatagrid/Column/AbstractColumn.php [hunk 3]
```diff
@@ -602,19 +669,43 @@ public function getRendererParameters($rendererName = 'jqGrid')
     }
 
     /**
-     * Set a a template formatter
+     * Set a template formatter and overwrite other formatter
+     *
+     * @param AbstractFormatter|AbstractFormatter[] $formatter
+     * @return $this
+     */
+    public function setFormatter($formatterList)
```
Comment (ThaDafinser) [inline src/ZfcDatagrid/Column/AbstractColumn.php]: Did u leave `setFormatter` for BC compability?

I would suggest to create a `setFormatters()` for multiple and leave that method for only one and deprecate it

File: src/ZfcDatagrid/Column/AbstractColumn.php [hunk 4]
```diff
@@ -602,19 +669,43 @@ public function getRendererParameters($rendererName = 'jqGrid')
...
     }
 
     /**
+     * return a list of different formatter
      *
-     * @param  string $rendererName
-     * @return NULL   AbstractFormatter
+     * @return AbstractFormatter[]
      */
     public function getFormatter()
```
Comment (ThaDafinser) [inline src/ZfcDatagrid/Column/AbstractColumn.php]: getFormatters()

File: src/ZfcDatagrid/Column/AbstractColumn.php [hunk 5]
```diff
@@ -623,7 +714,6 @@ public function getFormatter()
 
     /**
      *
-     * @param  string  $rendererType
      * @return boolean
      */
     public function hasFormatter()
```
Comment (ThaDafinser) [inline src/ZfcDatagrid/Column/AbstractColumn.php]: Maybe also with `s` (multiple)? but i'm not sure

**Comment by ThaDafinser** [comment]
Looks good so far with the BC compability keeped.

But i'm not sure with the fluent interface http://ocramius.github.io/blog/fluent-interfaces-are-evil/

**Comment by kokspflanze** [comment]
about the missing s, sure i can add it. but i dont think to remove the old methods without s, only mark them as deprecated and remove them in a later version.

about the fluent interface, it only return the own object, so its no high complexity, so i dont see a big problem =).

**Comment by ThaDafinser** [comment]
@kokspflanze i migrated/rebased to #174

**Pull request closed without merging**